src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -222,7 +226,11 @@ void Terminal::UpdateSettings(winrt::Microsoft::Terminal::Settings::ICoreSetting
         newViewportTop = oldRows.mutableViewportTop;
         newVisibleTop = oldRows.visibleViewportTop;
     }
-    CATCH_RETURN();
+    catch (...)
+    {
+        _buffer->GetCursor().EndDeferDrawing();

```

Comment:
In screenInfo, you said...
```c++
// Only tell _textBuffer to EndDefer (and not newTextBuffer) because whether or not Reflow failed,
// _textBuffer will end up being the buffer that we use after returning.
```
Should we just `CATCH_RETURN()` here then?

Also, if we return on line 221, would we be fine?

Reply:
So I wanted to make sure `_buffer` needs to EndDefer no matter what before we get out of this function. If the try failed at all and entered the catch, I need to call `EndDeferDrawing()` before we return the exception. 

I actually did not notice the `RETURN_IF_FAILED` for Reflow... Does that just totally skip the catch block? If so, I need to figure out where to EndDefer for `_buffer` 🤔

Reply:
> I actually did not notice the `RETURN_IF_FAILED` for Reflow... Does that just totally skip the catch block? If so, I need to figure out where to EndDefer for `_buffer` 🤔

From a glimpse at `RETURN_IF_FAILED`, yup. We'd return so the catch block never gets executed.

You could probably get around it with a `wil::scope_exit` though.

Reply:
Ok, I basically just separated out the `RETURN_IF_FAILED` call into its innards, but slapped an `EndDefer` call before it.

Reply:
I'd somewhat prefer a scope_exit if possible. I'm happy to merge this as-is if a scoped end-defer isn't easy, but i'm worried that we have _three_ different "end defer" locations here, and they refer to multiple different text buffers. idk

Reply:
Oh crap I'm so sorry @carlos-zamora 😢 , I didn't refresh my tab this morning when I replied to this thread so I didn't see the `scope_exit` comment before pushing my wacky change! Both of you are totally right, it's a much better approach 😅